src/DB/ProductFeedQueryHelper.php

```diff
@@ -94,14 +95,20 @@ public function get( WP_REST_Request $request ): array {
 		add_filter( 'posts_where', [ $this, 'title_filter' ], 10, 2 );
 
 		foreach ( $this->product_repository->find( $args, $limit, $offset ) as $product ) {
-			$id     = $product->get_id();
-			$errors = $product_helper->get_validation_errors( $product );
+			$id        = $product->get_id();
+			$errors    = $product_helper->get_validation_errors( $product );
+			$mc_status = $product_helper->get_mc_status( $product ) ?: $product_helper->get_sync_status( $product );
+
+			// If the refresh_status_data_job is scheduled, we don't know the status yet as it is being refreshed.
+			if ( $refresh_status_data_job && $refresh_status_data_job->is_scheduled() ) {
+				$mc_status = '-';
+			}
 
```

Comment:
❓  Do we really need this chunk of code? Maybe we can keep `status`as null when is refreshing.

Reply:
I think it is necessary because if there's no `mc_status`, it'll just return 'not-synced'. And most of the time, `mc_status` won't be null.

>If the mc_statuses is not set for the product, we will consider that product is not synced in the MC. See here: https://github.com/woocommerce/google-listings-and-ads/pull/2255#issuecomment-1941282154

https://github.com/woocommerce/google-listings-and-ads/blob/6b7a614b36fe71f0674025eda35e388209081d62/src/DB/ProductFeedQueryHelper.php#L100

I switched from using `-` to `null` because it might be clearer to indicate that there's no status yet.